rskj-core/src/main/java/co/rsk/rpc/modules/trace/TraceModuleImpl.java

```diff
@@ -166,4 +186,63 @@ private Block getByJsonBlockId(String id) {
             }
         }
     }
+
+    private List<TransactionTrace> buildBlockTraces(Block block, TraceFilterRequest traceFilterRequest) {
+        List<TransactionTrace> blockTraces = new ArrayList<>();
+
+        if (block.getNumber() != 0) {
+            List<Transaction> txList = block.getTransactionsList();
+
+            ProgramTraceProcessor programTraceProcessor = new ProgramTraceProcessor();
+
+            if (traceFilterRequest != null) {
```

Comment:
This method is too big for my liking, maybe you could extract some sub-methods to increase readability.

This if branch could be moved to a method like `txList = filter(block.getTransactionsList(), traceFilterRequest)`.

The `for (Transaction tx : txList) {` branch could be moved to another method like `transform(...)`.

WDYT?